confignode/src/main/java/org/apache/iotdb/confignode/service/executor/ConfigRequestExecutor.java

```diff
@@ -135,4 +148,56 @@ public TSStatus executorNonQueryPlan(ConfigRequest req)
         throw new UnknownPhysicalPlanTypeException(req.getType());
     }
   }
+
+  public boolean takeSnapshot(File snapshotDir) {
+
+    AtomicBoolean result = new AtomicBoolean(true);
+    getAllAttributes()
+        .parallelStream()
+        .forEach(
+            x -> {
+              boolean takeSnapshotResult = true;
+              try {
+                String fileName = x.getClass().getSimpleName();
```

Comment:
use class name as file name isn't a good idea. If we change the class name, then the old snapshot won't be load success.

Reply:
fixed.